### PR TITLE
Fix #82: Avoid creating a side-effect in typing module.

### LINKED ISSUE #82: cloudpickle breaks dill deserialization across servers.
Following up on this issue on Stackoverflow:

http://stackoverflow.com/questions/42960637/python-3-5-dill-pickling-unpickling-on-different-servers-keyerror-classtype/43006034#43006034

In a nutshell, with Python 3.5:

**Server A** imports `cloudpickle` this causes `types.ClassType` to become defined.
```
>>> import types
>>> dir(types)
  ['BuiltinFunctionType',
   'BuiltinMethodType',
   'ClassType',
   'CodeType',
   ...
  ]
```
**Server B** does not import `cloudpickle`, so `types.ClassType` is left undefined.
```
>>> import types
>>> dir(types)
  ['BuiltinFunctionType',
   'BuiltinMethodType',
   'CodeType',
   ...
  ]
```
Objects which are serialized in **server A** also seem to serialize a reference to `ClassType`. Then, when they are deserialized on **server B**, we encounter the following error:
```
Traceback (most recent call last):
 File "/home/streamsadmin/git/streamsx.topology/test/python/topology/deleteme2.py", line 40, in <module>
   a = dill.loads(base64.b64decode(a.encode()))
 File "/home/streamsadmin/anaconda3/lib/python3.5/site-packages/dill/dill.py", line 277, in loads
   return load(file)
 File "/home/streamsadmin/anaconda3/lib/python3.5/site-packages/dill/dill.py", line 266, in load
   obj = pik.load()
 File "/home/streamsadmin/anaconda3/lib/python3.5/site-packages/dill/dill.py", line 524, in _load_type
   return _reverse_typemap[name]
KeyError: 'ClassType'
```

I've found a workaround, which you can see on Stackoverflow. 

Here's my question: `types.ClassType` was removed in 3.5, yet cloudpickle re-adds it. Is this strictly necessary? It seems to be having side effects.

### PR DESCRIPTION
**Pull request by tvalentyn**
cc: @charlesccychen

### CODE REVIEW

**Comment by tvalentyn** [comment]
@pierreglaser @ogrisel - could you please take a look at this change? 

Somehow I heard from 3 different people about #82 in last two days, so I would appreciate if we can fix it in the next release. Thank you.

File: cloudpickle/cloudpickle.py [hunk 1]
```diff
@@ -95,7 +95,6 @@
     PY3 = False
     PY2 = True
 else:
-    types.ClassType = type
```
Comment (ogrisel) [inline cloudpickle/cloudpickle.py]: Wow that's wild! I had never realized that cloudpickle would do that...

**Comment by ogrisel** [comment]
@pierreglaser we should also run the downstream tests on this once your PR is ready.

**Comment by ogrisel** [comment]
I went ahead and updated the changelog. It think we should merge this quickly and release 1.3.0 directly soon after (but only once we have a change to run the downstream tests with the new CI config).

**Comment by ogrisel** [comment]
If we need we can always backport some fixes to 1.2.3.

**Comment by tvalentyn** [comment]
Thanks so much, @ogrisel !
I am curious about your CI setup - where do you follow the status of downstream tests?

**Comment by ogrisel** [comment]
> where do you follow the status of downstream tests?

Work in progress: #336

**Comment by ogrisel** [comment]
I tried to cancel the py35 ubuntu build that has been running for 30+ mins but it does not work. We cannot access the stdout of the pytest command... I don't know if it's a github actions bug or a real problem in cloudpickle.

Running locally the py35 tests on my linux box with tox completes in 10s...

**Comment by ogrisel** [comment]
Let's hope it was a bug in github actions...

**Comment by ogrisel** [comment]
Thanks @tvalentyn for the fix. Hopefully we will release 1.3.0 by next week.

**Comment by ogrisel** [comment]
Sorry, I merged before letting @pierreglaser have a look.

**Comment by tvalentyn** [comment]
No worries, I think one pair of eyes is sufficient on this change.

**Comment by tvalentyn** [comment]
Thank you!

**Pull request merged**